module/block_iterator/executor/executor.go

```diff
@@ -136,5 +135,37 @@ func commitAndCheckpoint(log zerolog.Logger, batch storage.Batch, iter module.Bl
 		return fmt.Errorf("failed to checkpoint iterator: %w", err)
 	}
 
+	startIndex, endIndex, nextIndex := iter.Progress()
+	progress := CalculateProgress(startIndex, endIndex, nextIndex)
+
+	log.Info().
+		Str("commit-dur", time.Since(start).String()).
+		Uint64("start-index", startIndex).
+		Uint64("end-index", endIndex).
+		Uint64("next-index", nextIndex).
+		Str("progress", fmt.Sprintf("%.2f%%", progress)).
+		Msg("batch committed")
+
+	metrics.ExecutionLastChunkDataPackPrunedHeight(nextIndex - 1)
+
 	return nil
 }
+
+// CalculateProgress calculates the progress of the iteration, it returns a percentage
+// of the progress. [0, 100]
+func CalculateProgress(start, end, current uint64) float64 {
+	if end <= start {
+		return 100.0 // If start == end, assume full completion
+	}
```

Comment:
I think block iterator `end` is inclusive, so `end == start` has one more block to iterate.  

```suggestion
	if end < start {
		return 100.0
	}
```